web-common/src/features/dashboards/leaderboard/leaderboard-widths.ts

```diff
@@ -0,0 +1,91 @@
+import { clamp } from "@rilldata/web-common/lib/clamp";
+import { writable } from "svelte/store";
+import type { LeaderboardItemData } from "./leaderboard-utils";
+
+const MIN_COL_WIDTH = 56;
+const DEFAULT_COL_WIDTH = 60;
+const MAX_COL_WIDTH = 132;
+
+export type ColumnWidths = {
+  dimension: number;
+  value: number;
+  percentOfTotal: number;
+  delta: number;
+  deltaPercent: number;
+};
+
+export const LEADERBOARD_DEFAULT_COLUMN_WIDTHS: ColumnWidths = {
+  dimension: 164,
+  value: DEFAULT_COL_WIDTH,
+  percentOfTotal: DEFAULT_COL_WIDTH,
+  delta: DEFAULT_COL_WIDTH,
+  deltaPercent: DEFAULT_COL_WIDTH,
+};
+
+// Create a store for the maximum column widths
+export const columnWidths = writable<ColumnWidths>(
+  LEADERBOARD_DEFAULT_COLUMN_WIDTHS,
+);
+export const processedDimensions = new Set<string>();
+
+export function updateMaxColumnWidths(
+  dimensionName: string,
+  newWidths: ColumnWidths,
+) {
+  if (!processedDimensions.has(dimensionName) && processedDimensions.size < 4) {
+    columnWidths.update((currentWidths) => ({
+      dimension: Math.max(currentWidths.dimension, newWidths.dimension),
+      value: Math.max(currentWidths.value, newWidths.value),
+      percentOfTotal: Math.max(
+        currentWidths.percentOfTotal,
+        newWidths.percentOfTotal,
+      ),
+      delta: Math.max(currentWidths.delta, newWidths.delta),
+      deltaPercent: Math.max(
+        currentWidths.deltaPercent,
+        newWidths.deltaPercent,
+      ),
+    }));
+    processedDimensions.add(dimensionName);
+  }
+}
```

Comment:
It could be simpler to make `updateMaxColumnWidths` a pure function that internalizes `processedDimensions` and returns `columnWidths`. It feels like `processedDimensions` is an implementation detail.

Reply:
Oh, now I realize I'm conflating widths within a single leaderboard & widths across all leaderboards. Maybe there's opportunity to more clearly distinguish between the two levels. And a pure function here could still help.